nft_ingester/src/program_transformers/bubblegum/db.rs

```diff
@@ -441,3 +441,327 @@ where
+
+    Ok(())
+}
+
+pub async fn unprotected_upsert_creators<T>(
+    txn: &T,
+    id: Vec<u8>,
+    creators: &Vec<Creator>,
+    slot_updated: i64,
+    seq: i64,
+) -> Result<(), IngesterError>
+where
+    T: ConnectionTrait + TransactionTrait,
+{
+    if !creators.is_empty() {
+        // Vec to hold base creator information.
+        let mut db_creator_infos = Vec::with_capacity(creators.len());
+
+        // Vec to hold info on whether a creator is verified.  This info is protected by `seq` number.
+        let mut db_creator_verified_infos = Vec::with_capacity(creators.len());
+
+        // Set to prevent duplicates.
+        let mut creators_set = HashSet::new();
+
+        for (i, c) in creators.iter().enumerate() {
+            if creators_set.contains(&c.address) {
+                continue;
+            }
+
+            db_creator_infos.push(asset_creators::ActiveModel {
+                asset_id: Set(id.clone()),
+                creator: Set(c.address.to_bytes().to_vec()),
+                position: Set(i as i16),
+                share: Set(c.share as i32),
+                slot_updated: Set(Some(slot_updated)),
+                ..Default::default()
+            });
+
+            db_creator_verified_infos.push(asset_creators::ActiveModel {
+                asset_id: Set(id.clone()),
+                creator: Set(c.address.to_bytes().to_vec()),
+                verified: Set(c.verified),
+                verified_seq: Set(Some(seq)),
+                ..Default::default()
+            });
+
+            creators_set.insert(c.address);
+        }
+
+        // This statement will update base information for each creator.
+        let query = asset_creators::Entity::insert_many(db_creator_infos)
+            .on_conflict(
+                OnConflict::columns([
+                    asset_creators::Column::AssetId,
+                    asset_creators::Column::Creator,
+                ])
+                .update_columns([
+                    asset_creators::Column::Position,
+                    asset_creators::Column::Share,
+                    asset_creators::Column::SlotUpdated,
+                ])
+                .to_owned(),
+            )
+            .build(DbBackend::Postgres);
+        txn.execute(query).await?;
+
+        // This statement will update whether the creator is verified and the
+        // `verified_seq` number.
+        let mut query = asset_creators::Entity::insert_many(db_creator_verified_infos)
+            .on_conflict(
+                OnConflict::columns([
+                    asset_creators::Column::AssetId,
+                    asset_creators::Column::Creator,
+                ])
+                .update_columns([
+                    asset_creators::Column::Verified,
+                    asset_creators::Column::VerifiedSeq,
+                ])
+                .to_owned(),
+            )
+            .build(DbBackend::Postgres);
+        query.sql = format!(
```

Comment:
Why is the method called "unprotected" if this case exists?

Reply:
Misnomer.  I guess it should have been called "partially protected" or something.  The protection on the verified flag was needed for verify/unverify.  But the positions, share amounts, etc. would be blanket protected by `update_metadata_seq`.